distributed/scheduler.py

```diff
@@ -2212,18 +2212,17 @@ def decide_worker_non_rootish(self, ts: TaskState) -> WorkerState | None:
             wp_vals = cast("Sequence[WorkerState]", worker_pool.values())
             n_workers: int = len(wp_vals)
             if n_workers < 20:  # smart but linear in small case
-                ws = min(wp_vals, key=operator.attrgetter("occupancy"))
+                ws = min(wp_vals, key=self.worker_objective_no_deps)
```

Comment:
This is a small behavioral change: root-ish tasks won't have affinity for workers that hold their dependencies. By definition, root-ish tasks shouldn't consider the location of their dependencies when scheduling, so this is arguably an improvement—it could slightly reduce dogpiling towards an open Zarr dataset, for instance.